File: 프로그래머스/1/135808. 과일 장수/과일 장수.py
def solution(k, m, score):
    answer = 0
    
    score.sort (reverse= True)

    
    can_make_box = len(score) // m
    idx=0
    while can_make_box >0 :
        target = score[idx:idx+m]
        min_val = target[m-1]
        answer += (min_val *m)
        idx =idx +m
        can_make_box -=1
            
    return answer

File: 프로그래머스/1/135808. 과일 장수/과일 장수.java
import java.util.*;
class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        
        ArrayList <Integer>  li = new ArrayList <>();
        
        for (int s : score) {
            li.add (s);
        }
        
        Collections.sort (li, Collections.reverseOrder());
        int box_cnt = (score.length /m);
        int idx =0;
        
        while (box_cnt >0) {
            
           answer += (li.get (idx+m-1) * m);
           idx =idx +m;
            box_cnt -=1;
        }
        
        return answer;
    }
}